### ISSUE TITLE
🚀 Feature Request: Make image size responsive

### ISSUE THREAD

**Issue by pauserratgutierrez** [elementor/elementor#18427]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

We can't set different image sizes for different devices.
![image](https://user-images.githubusercontent.com/73240206/162963617-b84f4321-adf5-4680-9818-3027b3a78343.png)

### Describe the solution you'd like

Make the image size feature responsive like other Elementor options are.

### Describe alternatives you've considered

Custom css

**Comment by Dcfranklyn** [comment]
I would like this idea as well. 
- Please add to all image related widgets/elements

**Comment by pauserratgutierrez** [comment]
@rami-elementor

**Comment by rami-elementor** [comment]
First of all, **images in Elementor are responsive**. If you look at the markup, you'll notice that the size you choose is assigned to **src** attribute, while all other sizes are set to **srcset** attribute, which aids the browser in selecting the optimum image quality to display.

```html
<img
 width="800"
 height="534"
 alt="Placeholder image"
 class="attachment-large size-large"
 loading="lazy"
 src="http://devedition.local/wp-content/uploads/2022/04/placeholder-1024x683.png"
 srcset="
  http://devedition.local/wp-content/uploads/2022/04/placeholder-1024x683.png 1024w,
  http://devedition.local/wp-content/uploads/2022/04/placeholder-300x200.png 300w,
  http://devedition.local/wp-content/uploads/2022/04/placeholder-768x512.png 768w,
  http://devedition.local/wp-content/uploads/2022/04/placeholder.png 1200w"
 sizes="(max-width: 800px) 100vw, 800px"
>
```

----

I believe what you're requesting is for the "Image Size Control" to be a responsive control to allow you to choose multiple sizes for desktop, tablet, and mobile views. This isn't possible with images since the URL is set in HTML; it's only possible with background-images because the URLs are set using CSS.

It's not an Elementor limitation, but rather an HTML/CSS limitation.

**Comment by Dcfranklyn** [comment]
@rami-elementor
Hello,

I understand why it's a code limitation for Elementor now

**Quick question:** Would it be possible the change not the image size but the image ratio for the multiple screen sizes (desktop, tablet, and mobile views). 

Example:
- Leave the same 800 x 534 for desktop
- Change to 534 x 534 for mobile

**Comment by rami-elementor** [comment]
@Dcfranklyn 

The image widget let's you set the width and height for each break point.

**Comment by Dcfranklyn** [comment]
@rami-elementor
Thanks for the quick reply. Your insight is very helpful :D

Sorry, I understand the image widget has these settings, but that sadly isn't the only way we use images within elementor. The same breakpoint feature set isn't within the Gallery, Basic Gallery, Image box, Image carousel, etc.

That's what I meant by "Please add to all image related widgets/elements"

There are loopholes - like creating two widgets with different settings and then hiding/showing the correct widget based on the breakpoint. This creates extra work/code that could be avoided if the features were available.

Would be lovely if each image widget had
- Height & Width (max width/height as well)
- Aspect Ratio (when not using custom image size)
- Object fit (Fill, cover, contain)
- Image spacing/gap (Galleries/carousels)
- Layout with breakpoint settings (Grid, justified, masonry)
- Etc

Every setting possible with breakpoint settings, so you can decide what settings need to change for desktop vs mobile. I'm aware many of these settings can be adjusted with custom CSS or loopholes, but that is more work of course.

I'm sure the Elementor team has considered many of these settings/features already. Some or many might not even be possible at this time. Just saying from a designer perspective the hurdles that I run into when dealing with images.

**Comment by rami-elementor** [comment]
@Dcfranklyn Thank you for taking the time to provide such detailed feedback. Our product management have been notified.

**Comment by arielk** [comment]
I merged this issue with another issue https://github.com/elementor/elementor/issues/6778 in the same subject so we could deal with it more efficiently. So please let's continue the conversation there.

Thanks for your feedback!

**Comment by nicholaszein** [comment]
Hello everyone!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor **v3.11**! 🥳 

✔️ Please feel free to check it out and update your plugin to the new version!

Cheers 🥂
